### PR TITLE
Logic change command bug fix

### PR DESCRIPTION
**Pull request by shaunthium**

### CODE REVIEW

File: deadline_tasks.json [hunk 1]
```diff
@@ -0,0 +1 @@
+[{"dateTime":{"date":{"year":2016,"month":3,"day":29},"time":{"hour":21,"minute":0,"second":0,"nano":0}},"completed":true,"name":"Work on Capstone Project"},{"dateTime":{"date":{"year":2016,"month":3,"day":30},"time":{"hour":15,"minute":0,"second":0,"nano":0}},"completed":true,"name":"Work on 2010 Lab"}]
```
Comment (tran-tien-dat) [inline deadline_tasks.json]: mistakes were made

File: events.json [hunk 2]
```diff
@@ -0,0 +1 @@
...cond":0,"nano":0}},"endDateTime":{"date":{"year":2016,"month":3,"day":30},"time":{"hour":15,"minute":0,"second":0,"nano":0}}}],"name":"Working at VersaFleet"},{"dateTimes":[{"startDateTime":{"date":{"year":2016,"month":3,"day":30},"time":{"hour":19,"minute":0,"second":0,"nano":0}},"endDateTime":{"date":{"year":2016,"month":3,"day":30},"time":{"hour":19,"minute":30,"second":0,"nano":0}}}],"name":"Discussion of CS2105 Assignment 4"},{"dateTimes":[{"startDateTime":{"date":{"year":2016,"month":3,"day":31},"time":{"hour":9,"minute":0,"second":0,"nano":0}},"endDateTime":{"date":{"year":2016,"month":3,"day":31},"time":{"hour":18,"minute":0,"second":0,"nano":0}}}],"name":"Working at VersaFleet"},{"dateTimes":[{"startDateTime":{"date":{"year":2016,"month":4,"day":2},"time":{"hour":9,"minute":0,"second":0,"nano":0}},"endDateTime":{"date":{"year":2016,"month":4,"day":3},"time":{"hour":22,"minute":0,"second":0,"nano":0}}}],"name":"Meeting"},{"dateTimes":[{"startDateTime":{"date":{"year":2016,"month":7,"day":2},"time":{"hour":9,"minute":0,"second":0,"nano":0}},"endDateTime":{"date":{"year":2016,"month":7,"day":1},"time":{"hour":18,"minute":0,"second":0,"nano":0}}}],"name":"Work at VersaFleet"}]
```
Comment (tran-tien-dat) [inline events.json]: mistakes were made

File: floating_tasks.json [hunk 3]
```diff
@@ -0,0 +1 @@
+[{"completed":true,"name":"And Let Me Play Amongst The Stars"},{"completed":true,"name":"Fly Me To The Moon"},{"completed":false,"name":"I love you"},{"completed":false,"name":"In Other Words"},{"completed":false,"name":"In other words"},{"completed":false,"name":"Let Me See What Spring is like on Jupiter and Mars"},{"completed":false,"name":"Plan module mapping for NOC by tonight 9pm"},{"completed":false,"name":"Please be true"},{"completed":true,"name":"Work on 2010 Assignment"}]
```
Comment (tran-tien-dat) [inline floating_tasks.json]: mistakes were made

File: save_data.json [hunk 4]
```diff
@@ -0,0 +1,44 @@
+{
```
Comment (tran-tien-dat) [inline save_data.json]: mistakes were made

**Comment by tran-tien-dat** [comment]
How about floating task? should you apply this approach to it as well for consistency?

**Comment by shaunthium** [comment]
@tran-tien-dat what do you mean by apply same approach to floating task?

**Comment by tran-tien-dat** [comment]
Calling its own constructor instead of the parent's

**Pull request merged**